app/assets/javascripts/uploadcare/settings.coffee

```diff
@@ -147,6 +149,21 @@ uploadcare.namespace 'settings', (ns) ->
     quality: shrink[3] / 100 if shrink[3]
     size: size
 
+  defaultPreviewUrlCallback = (url, info) ->
+    return url if not @previewProxy
+    
+    encodedUrl = encodeURIComponent(url)
+
+    justAppend = /\=$/.test(@previewProxy)
+    useAmpersand = /[^\&\?\=]$/.test(@previewProxy)
+    addQuestionSign = not /\?/.test(@previewProxy)
+
+    queryPart = if justAppend then encodedUrl else "url=#{encodedUrl}"
+    queryPart = '&'.concat(queryPart) if useAmpersand
+    queryPart = '?'.concat(queryPart) if addQuestionSign
```

Comment:
In the rare case when the URL end with `=` and don't have `?` you'll get:
`http://domain.com/=?https%3A%2F%2Fcdn.yourdomain.com%2F{uuid}%2F `.

Reply:
@homm yep, but who will do this? I think we should cover the most common input cases. And `http://domain.com/=` is not such case. No one will do that.

Reply:
I'm not thinking in terms of probability. I think in terms of core correctness. The code should do what it was designed for. There was a bunch of other problems, see comment to the last commit: https://github.com/uploadcare/uploadcare-widget/pull/454/commits/03fb0d1c6e4337b34bb1640913ef866b29bf23a6

Reply:
The most significant was URL normalization, which trims trailing slash.